app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryimage/EntryImageActivity.java

```diff
@@ -0,0 +1,153 @@
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_entry_image;
+    }
+
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+        if (!getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
+            chooseCameraButton.setEnabled(false);
+        }
+    }
+
+    @Override
+    public String getEntryName() {
+        return getIntent().getStringExtra(EXTRA_ENTRY_NAME);
+    }
+
+    @Override
+    public void showEntryName(String entryName) {
+        previewLabelTextView.setText(getResources().getString(R.string.entry_image_preview_caption, entryName));
+    }
+
+    @Override
+    public void displayChooseImageLayout() {
+        chooseImageLayout.setVisibility(android.view.View.VISIBLE);
+        chooseImageLayout.bringToFront();
+        previewImageLayout.setVisibility(android.view.View.INVISIBLE);
+    }
+
+    @Override
+    public void displayPreviewImageLayout(Bitmap bitmap) {
+        chooseImageLayout.setVisibility(android.view.View.INVISIBLE);
+        previewImageLayout.setVisibility(android.view.View.VISIBLE);
+        previewImageLayout.bringToFront();
+        previewImageView.setImageBitmap(bitmap);
+    }
+
+    @OnClick(R.id.entry_image_camera_button)
+    protected void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+        }
+    }
+
+    @OnClick(R.id.entry_image_gallery_button)
+    protected void dispatchChoosePictureIntent() {
+        String pickImageMessage = getResources().getString(R.string.pick_image_title);
+
+        Intent androidGalleryIntent = new Intent(Intent.ACTION_GET_CONTENT);
+        androidGalleryIntent.setType(PICK_IMAGE_TYPE);
+
+        Intent externalGalleriesIntent = new Intent(Intent.ACTION_PICK,
+                                                    android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        externalGalleriesIntent.setType(PICK_IMAGE_TYPE);
+
+        Intent chooserIntent = Intent.createChooser(androidGalleryIntent, pickImageMessage);
+        chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[] { externalGalleriesIntent });
+
+        startActivityForResult(chooserIntent, REQUEST_PICK_IMAGE);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode != RESULT_OK || (requestCode != REQUEST_IMAGE_CAPTURE && requestCode != REQUEST_PICK_IMAGE)) {
+            super.onActivityResult(requestCode, resultCode, data);
+            return;
+        }
+
+        Bitmap bitmap = null;
+
+        if (requestCode == REQUEST_IMAGE_CAPTURE) {
+            Bundle extras = data.getExtras();
+            bitmap = (Bitmap) extras.get("data");
```

Comment:
Makes sense! Later the value (or the approach) might change because this is getting the thumbnail, while I figure out how to get the full-size image (or at least a bigger thumbnail!)